### DISCUSSION TITLE
How to access texts field of foxglove.ImageAnnotations?

### DISCUSSION THREAD

**Discussion by Prokrus** [foxglove/community#716] [Q&A] [answered]
In document, there is a **texts** filed in ImageAnnotations, but when I declare a ImageAnnotation in user script, there is no texts field in it.
![image](https://github.com/foxglove/community/assets/26819180/225bb177-1b81-4de9-8b5b-49c0bdcaf1b7)

**Comment by defunctzombie** [comment]
@Prokrus Which version of Studio?

**Reply by Prokrus** [reply]
1.53.1

**Comment by defunctzombie** [comment]
The autocomplete behavior for the `Message<>` comes from the data source you load. So if the datasource you load describes a `foxglove.ImageAnnotation` as a message with only `circles` and `points` fields then that's all you will see in the autocomplete. If you want to have the `text` field in the autocomplete you'll need to open a data source that has an updated `foxglove.ImageAnnotation` schema.

**Reply by Prokrus** [reply]
It's no only autocomplete, even if I create a new Message<foxglove.ImageAnnotation>, it contains circles and points field but no texts filed.
I want to add texts message based on data source and display it. Do you know how can I achieve that? Thank you.

**Reply by defunctzombie** [reply]
Thanks for the feedback. We've identified some improvements we can make for the workflow you describe and will follow up once we have made those improvements.

**Comment by foxymiles** [comment, marked as answer]
In the latest version of studio you can now import and use types from the `@foxglove/schemas` package, so you should be able to do something like this:

```typescript
import { Input } from "./types";
import { ImageAnnotations } from "@foxglove/schemas";

export const inputs = ["/imu"];
export const output = "/studio_script/output_topic";

export default function script(event: Input<"/imu">): ImageAnnotations {
  return {
    circles: [],
    points: [],
    texts: [],
  };
}
```

This should work now on https://studio.foxglove.dev and on the desktop app in our next release.

We'll have documentation on this very soon.
